Add manage data section to Promscale (#678)
timescale/docs: promscale/manage-data/downsampling.md

```sql
SELECT register_metric_view('public', 'node_memfree_1hour');
```

You can now use this data as a regular metric in both SQL and PromQL.

To query the average in the new aggregated metric with SQL, and show it in a
time series chart in Grafana, use this query:
```sql
SELECT time, jsonb(labels) as metric, avg
FROM node_memfree_1hour
WHERE $__timeFilter(time)
ORDER BY time asc
```

To do the same with PromQL, you need to use the `__column__` label. The
aggregated metric stores the data for `min`, `max`, and `avg` in different
columns, which is not supported by Prometheus, because Prometheus stores all
metrics independently. Promscale uses the `__column__` label to identify which
column to return. If no `__column__` label is specified, Promscale returns the
`value` column by default.  To query the average in the new aggregated metric
with PromQL, use this query:
```promql
node_memfree_1hour{__column__="avg"}
```

You do not need to specify the `__column__` label in queries, because Promscale
uses the value column by default when ingesting data.

You could modify this continuous aggregate to name one of the metrics `value`,

> which would them make Promscale return that metric even if no `__column__` label

Reviewer: ```suggestion
which would then make Promscale return that metric even if no `__column__` label
```

is specified. To do this with the previous example, you can create a continuous
aggregate that sets the average as the default to be returned in PromQL queries,
like this:
```sql

> CREATE MATERIALIZED VIEW node_memfree_1hour( time, min, max, avg, count, sum, series_id)

Reviewer 2: @VineethReddy02 @ramonguiu - please confirm.

```sql
{__name__=~"node_mem*"} // this valid PromQL query will not match our previously created metric view
```

</highlight>

### Deleting continuous aggregates
To delete a Promscale continuous aggregate, delete the metric view first, and
then remove the continuous aggregate.

Remove the metric view:
```sql
unregister_metric_view('public', 'node_memfree_1hour');
```

Delete the continuous aggregate:
```sql
DROP MATERIALIZED VIEW node_memfree_1hour;
```

## Prometheus downsampling with recording rules
Prometheus provides a downsampling mechanism called recording rules. These rules
are used to calculate frequently used or computationally expensive queries
before they are required. A recording rule is a PromQL expression that
Prometheus evaluates at a predefined frequency to generate a new metric series.
The new metric series is stored in Promscale, and you can query it in the same
way as any other Prometheus metric. An example recording rule:

> ```sql

Reviewer 3: ```suggestion
```yaml
```

groups:
  - name: daily_stats
    interval: 1h
    rules:
    - record: customer:api_requests:rate1day
      expr: sum by (customer) (rate(api_requests_total[1d]))

```

For the recording rules to be applied, add the location of the recording rules
file to the Prometheus configuration file:
```yaml
rule_files:

>   - "<recording-rules-file>"

Reviewer 3: ```suggestion
  - "<RECORDING-RULES-FILE>"
```

```

<highlight type="note">
When you set up the Prometheus configuration file to use recording rules, we
recommend that you also set `read_recent` to `true` in the
[Prometheus remote_read configuration](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#remote_read).
This tells Prometheus to fetch data from Promscale when evaluating PromQL
queries, including recording rules. If `read_recent` is disabled, only the data

> stored in the local TimescaleDB instance is used when evaluating alerting and

Author: This originally said "data stored in Prometheus's local tsdb". Tsdb in this context is not TimescaleDB, it's how the local timeseries database is named in Prometheus: https://prometheus.io/docs/prometheus/latest/storage/ (search for tsdb).

To disambiguate, I suggest we say something like "stored in Prometheus' local on-disk time series database"

Reviewer: ```suggestion
stored in the Prometheus local on-disk database is used when evaluating alerting and
```

recording rules, so it is dependent on the retention period of Prometheus rather
than Promscale.
</highlight>

To query the metric with PromQL:
```promql
customer:api_requests:rate1day
```

To query the metric with SQL:
```sql
SELECT time, jsonb(labels) as metric, value
FROM "customer:api_requests:rate1day"

> ORDER BY time asc

Reviewer 3: ```suggestion
ORDER BY time ASC
```

# Downsample data in Promscale
Downsampling is the term used when reducing the rate of a signal. When you
downsample data, the size of the data on disk is reduced, but so is the
resolution of the data. This means that you have fewer entries in your source
data, to take up less space on disk. The main reasons for downsampling are to
reduce cost and increase performance. Querying downsampled data gets faster as
the size of the data decreases.

## Choose a downsampling method
There are two downsampling methods available for use with Promscale: continuous

> aggregates and Prometheus recording rules.

Reviewer 4: aggregates, and

> aggregates and Prometheus recording rules.

Reviewer 4: ```suggestion
aggregates, and Prometheus recording rules.
```

*   Backfilling: continuous aggregates automatically downsample all data
    available, including past data, so that performance improvements on the
    aggregated metric are seen as soon as it is created.  

For Promscale to be able to use a continuous aggregate as a metric view, the
continuous aggregate must:

*   Be based on a raw metric series ingested by Promscale that is specified in
    the `FROM` clause.
*   Include a column named `time` of type `timestamptz`, that corresponds to the
    time associated with each aggregated metric sample.
*   Include a column named `series_id` of type `bigint`,  that corresponds to
    the `series_id`  from the raw metrics.
*   Any number of additional columns of type `double precision`, that correspond
    to the metric values you want to store.

<highlight type="important">
Continuous aggregates supports only one metric in the `FROM` clause, and can
generate aggregations only within the same `series_id`. This is because
the `series_id` corresponds to a specific metric name and set of labels.
</highlight>

For more information about continuous aggregates, see the
[TimescaleDB continuous aggregates documentation][tsdb-caggs].

> ### Worked example of a Promscale continuous aggregate

Reviewer 4: Example of a Promscale.....

> ### Worked example of a Promscale continuous aggregate

Reviewer 4: ```suggestion
### Example of a Promscale continuous aggregate
```

Creating a continuous aggregate in Promscale requires two operations. Start by
creating a TimescaleDB continuous aggregate, and then register the new metric so
it's available to PromQL queries and other Promscale functions. This section
includes a worked example.

This example uses a metric called `node_memory_MemFree`, and shows you how to
create a continuous aggregate to derive some summary statistics about the metric
on an hourly basis. Run this query on the underlying TimescaleDB database:
```sql

> CREATE MATERIALIZED VIEW node_memfree_1hour( time, min, max, avg, count, sum, series_id)

Reviewer 2: ```suggestion
CREATE MATERIALIZED VIEW node_memfree_1hour( time, series_id, min, max, avg)
```

WITH (timescaledb.continuous) AS
  SELECT
        time_bucket('1 hour', time) + '1 hour' as time ,
        series_id,
        min(value) as min,
        max(value) as max,
        avg(value) as avg
    FROM prom_data.node_memory_MemFree
    GROUP BY time_bucket('1 hour', time), series_id
```

<highlight type="note">

> In the worked example in this section, one hour is added to `time_bucket` in the

Reviewer 4: In this example, one hour is .....

> In the worked example in this section, one hour is added to `time_bucket` in the

Reviewer 4: ```suggestion
In this example, one hour is added to `time_bucket` in the
```
